druid/src/widget/tabs.rs

```diff
@@ -0,0 +1,960 @@
+        tabs.push(InitialTab::new(name, child))
+    }
+}
+
+#[derive(Clone, Lens, Data)]
+pub struct TabsState<TP: TabsPolicy> {
+    pub inner: TP::Input,
+    pub selected: TabIndex,
+    pub policy: TP,
+}
+
+impl<TP: TabsPolicy> TabsState<TP> {
+    pub fn new(inner: TP::Input, selected: usize, policy: TP) -> Self {
+        TabsState {
+            inner,
+            selected,
+            policy,
+        }
+    }
+}
+
+pub struct TabBar<TP: TabsPolicy> {
+    axis: Axis,
+    cross: CrossAxisAlignment,
+    orientation: TabsOrientation,
+    tabs: Vec<(TP::Key, TabBarPod<TP>)>,
+    hot: Option<TabIndex>,
+    phantom_tp: PhantomData<TP>,
+}
+
+impl<TP: TabsPolicy> TabBar<TP> {
+    pub fn new(axis: Axis, cross: CrossAxisAlignment, orientation: TabsOrientation) -> Self {
+        TabBar {
+            axis,
+            cross,
+            orientation,
+            tabs: vec![],
+            hot: None,
+            phantom_tp: Default::default(),
+        }
+    }
+
+    fn find_idx(&self, pos: Point) -> Option<TabIndex> {
+        let major_pix = self.axis.major_pos(pos);
+        let axis = self.axis;
+        let res = self
+            .tabs
+            .binary_search_by_key(&((major_pix * 10.) as i64), |(_, tab)| {
+                let rect = tab.layout_rect();
+                let far_pix = axis.major_pos(rect.origin()) + axis.major(rect.size());
+                (far_pix * 10.) as i64
+            });
+        match res {
+            Ok(idx) => Some(idx),
+            Err(idx) if idx < self.tabs.len() => Some(idx),
+            _ => None,
+        }
+    }
+
+    fn ensure_tabs(&mut self, data: &TabsState<TP>) {
+        // Borrow checker/ type inference fun
+        let (orientation, axis, cross) = (self.orientation, self.axis, self.cross);
+        let finish_row = |w| WidgetPod::new(orientation.rotate_and_box(w, axis, cross));
+        let finish_label = |w| WidgetPod::new(orientation.rotate_and_box(w, axis, cross));
+
+        ensure_for_tabs(&mut self.tabs, &data.policy, &data.inner, |policy, key| {
+            let info = policy.tab_info(key.clone(), &data.inner);
+
+            let label = data
+                .policy
+                .tab_label(key.clone(), &info, &data.inner)
+                // TODO: Type inference fails here because both sides of the lens are dependent on
+                // associated types of the policy. Needs changes to lens derivation to embed PhantomData of the (relevant?) type params)
+                // of the lensed types into the lens, to type inference has something to grab hold of
+                .lens::<TabsState<TP>, tabs_state_derived_lenses::inner>(TabsState::<TP>::inner)
+                .padding(Insets::uniform_xy(9., 5.));
+
+            if info.can_close {
+                let row = Flex::row()
+                    .with_child(label)
+                    .with_child(Label::new("ⓧ").on_click(
```

Comment:
On my configuration (GTK, but unsure what font druid is using), this doesn't show up.

Reply:
I noticed that too on Linux (when working out rzhks issues). I guess the options are to use images or paint it - probably worth doing the latter I suppose.

Reply:
I think I'd prefer to do this in a follow up as it might get a bit involved to make this button work well.

Reply:
my inclination would be to paint this, but follow-up is fine. maybe we do something gross for now, like just use 'X' as the text?

Reply:
Promise I'll do a follow up pretty quick...